### PR TITLE
[WIP] KeyVault Track 2 SDK adoption

### PR DESCRIPTION
**Pull request by bim-msft**
WIP: DO NOT MERGE

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by yungezz** [comment]
this PR is experiment for Track 2 SDK generation, pls do NOT review or merge. thanks.

File: sdk/keyvault/azure-mgmt-keyvault/azure/mgmt/keyvault/_version.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+VERSION = "0.1.0"
```
Comment (lmazuel) [inline sdk/keyvault/azure-mgmt-keyvault/azure/mgmt/keyvault/_version.py]: 3.0.0b1 ?

File: sdk/keyvault/azure-mgmt-keyvault/setup.py [hunk 2]
```diff
@@ -81,6 +81,7 @@
         'msrest>=0.5.0',
         'msrestazure>=0.4.32,<2.0.0',
```
Comment (lmazuel) [inline sdk/keyvault/azure-mgmt-keyvault/setup.py]: you don't need msrestazure anymore

File: sdk/keyvault/azure-mgmt-keyvault/setup.py [hunk 3]
```diff
@@ -81,6 +81,7 @@
...
         'azure-common~=1.1',
+        'azure-mgmt-core'
```
Comment (lmazuel) [inline sdk/keyvault/azure-mgmt-keyvault/setup.py]: azure-mgmt-core>=1.0.0b1,<2.0.0

File: sdk/keyvault/azure-mgmt-keyvault/setup.py [hunk 4]
```diff
@@ -81,6 +81,7 @@
         'msrest>=0.5.0',
```
Comment (lmazuel) [inline sdk/keyvault/azure-mgmt-keyvault/setup.py]: msrest>0.6.11 required for track2

File: sdk/keyvault/azure-mgmt-keyvault/azure/mgmt/keyvault/_version.py (hunk 1 above)
Comment (bim-msft) [reply sdk/keyvault/azure-mgmt-keyvault/azure/mgmt/keyvault/_version.py]: > 3.0.0b1 ?

Thanks, I will bump the version here.

File: sdk/keyvault/azure-mgmt-keyvault/setup.py (hunk 4 above)
Comment (bim-msft) [reply sdk/keyvault/azure-mgmt-keyvault/setup.py]: > msrest>0.6.0, or better msrest>0.6.11 (even if technically it's just pushing bug fixes and is not _necessary_)

Thanks, I will follow your suggestion.

**Comment by iscai-msft** [comment]
hey @bim-msft can you regenerate using https://github.com/Azure/autorest.python/releases/tag/v5.0.0-dev.20200326.1?

This will remove the "azure-" prefix from the user agent name to fulfill library guidelines (also includes other fixes, not sure how / if they apply here)

Thanks so much!

**Comment by yungezz** [comment]
> hey @bim-msft can you regenerate using https://github.com/Azure/autorest.python/releases/tag/v5.0.0-dev.20200326.1?
> 
> This will remove the "azure-" prefix from the user agent name to fulfill library guidelines (also includes other fixes, not sure how / if they apply here)
> 
> Thanks so much!

Thanks @iscai-msft for the info. Just a question about userAgent in general, should `azure-` prefix been removed from all languages' SDKs? thanks!

**Comment by bim-msft** [comment]
> hey @bim-msft can you regenerate using https://github.com/Azure/autorest.python/releases/tag/v5.0.0-dev.20200326.1?
> 
> This will remove the "azure-" prefix from the user agent name to fulfill library guidelines (also includes other fixes, not sure how / if they apply here)
> 
> Thanks so much!

@iscai-msft Thanks, I will have a try.

File: sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/models/_models.py [hunk 5]
```diff
@@ -547,23 +548,29 @@ def __init__(
...
 
+    Variables are only populated by the server, and will be ignored when sending a request.
+
     :param private_endpoint: Properties of the private endpoint object.
     :type private_endpoint: ~azure.mgmt.keyvault.v2018_02_14.models.PrivateEndpoint
     :param private_link_service_connection_state: Approval state of the private link connection.
     :type private_link_service_connection_state:
      ~azure.mgmt.keyvault.v2018_02_14.models.PrivateLinkServiceConnectionState
-    :param provisioning_state: Provisioning state of the private endpoint connection. Possible
+    :ivar provisioning_state: Provisioning state of the private endpoint connection. Possible
```
Comment (iscai-msft) [inline sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/models/_models.py:564]: Did anything change in the swagger in your regeneration for `provisioning_state` to become readonly?

File: sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/aio/operations_async/_private_endpoint_connections_operations_async.py [hunk 6]
```diff
@@ -119,16 +120,21 @@ def __init__(self, client, config, serializer, deserializer) -> None:
         :param private_endpoint_connection_name: Name of the private endpoint connection associated
          with the key vault.
         :type private_endpoint_connection_name: str
-        :param properties: The intended state of private endpoint connection.
-        :type properties: ~azure.mgmt.keyvault.v2018_02_14.models.PrivateEndpointConnection
+        :param private_endpoint: Properties of the private endpoint object.
```
Comment (iscai-msft) [inline sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/aio/operations_async/_private_endpoint_connections_operations_async.py:123]: Just want to make sure: when you regenerated this time, did you specify a payload flattening threshold? Did your command change in anyways?

File: sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/models/_models.py (hunk 5 above)
Comment (bim-msft) [reply sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/models/_models.py:564]: > Did anything change in the swagger in your regeneration for `provisioning_state` to become readonly?

@iscai-msft I didn't change anything on swagger, is this an unexpected change?

File: sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/aio/operations_async/_private_endpoint_connections_operations_async.py (hunk 6 above)
Comment (bim-msft) [reply sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/aio/operations_async/_private_endpoint_connections_operations_async.py:123]: > Just want to make sure: when you regenerated this time, did you specify a payload flattening threshold? Did your command change in any ways?

@iscai-msft No, I didn't change my regen command.
```
autorest --python --use=/mnt/c/repos/autorest.python /mnt/c/repos/azure-rest-api-specs/specification/keyvault/resource-manager/readme.md --multiapi --python-sdks-folder=/mnt/c/repos/azure-sdk-for-python/sdk --debug
```

File: sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/models/_models.py (hunk 5 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-mgmt-keyvault-autorestv5/azure/mgmt/keyvault/v2018_02_14/models/_models.py:564]: Ok I see, this is my bad. We made a fix where readonly being defined in the object definition propagates up so the property becomes readonly as well. Sorry for the confusion

**Comment by bim-msft** [comment]
Closing it due to we have a larger coverage now: #11484

**Pull request closed without merging**